frontend/src/components/DocumentClassPage.js

```diff
@@ -0,0 +1,186 @@
+  }
+
+  async componentDidMount() {
+    const res = await getAllDocumentClasses()
+    if (res) {
+      this.props.updateDocumentClasses(res)
+    } else {
+      this.props.updateDocumentClasses([])
+    }
+  }
+
+  toggle() {
+    this.setState({ addModal: !this.state.addModal })
+  }
+
+  secondToggle() {
+    this.setState({ uploadModal: !this.state.uploadModal })
+  }
+
+  updateName = event => {
+    this.setState({ name: event.target.value })
+  }
+
+  updateDescription = event => {
+    this.setState({ description: event.target.value })
+  }
+
+  handleSubmit() {
+    createDocumentClass(
+      this.state.name,
+      this.state.description,
+      this.state.files[0],
+      this.state.files[0].name
+    )
+    this.secondToggle()
+  }
+
+  onDrop(files) {
+    this.setState({
+      files
+    })
+  }
+
+  render() {
+    return (
+      <>
+        <Modal isOpen={this.state.uploadModal} toggle={this.toggleUpload}>
+          <Upload uploadType="DocumentClass" />
+        </Modal>
+        <Modal isOpen={this.state.addModal} toggle={this.toggle}>
+          <ModalBody>
+            <form>
+              <span> Name: </span>
+              <input onChange={this.updateName} />
+              <br />
+              <span> Description: </span>
+              <textarea
+                name="paragraph_text"
+                cols="50"
+                rows="10"
+                onChange={this.updateDescription}
+              />
+              <br />
+              <div className="dropPage">
+                <section className="droppedBox">
+                  <div className="dropZone">
+                    <Dropzone onDrop={this.onDrop.bind(this)}>
+                      {({ getRootProps, getInputProps }) => (
+                        <section>
+                          <div {...getRootProps()}>
+                            <input {...getInputProps()} />
+                            <p>Drag some files here, or click to select files</p>
+                          </div>
+                        </section>
+                      )}
+                    </Dropzone>
+                  </div>
+                  <aside>
+                    <h4>Files Dropped</h4>
+                    <ul className="droppedFilesBackground">
+                      {this.state.files.map(f => (
+                        <li className="droppedBox" key={f.name}>
+                          {f.name} - {f.size} bytes
+                        </li>
+                      ))}
+                    </ul>
+                    <Button
+                      disabled={this.state.files.length === 0}
+                      className="right"
+                      onClick={this.handleSubmit}
+                    >
+                      Create Document Class
+                    </Button>
+                    <Modal isOpen={this.state.uploadModal} toggle={this.secondToggle}>
+                      <ModalBody>File uploaded - your submission is being processed.</ModalBody>
```

Comment:
Im wondering if there's a need for nested modals - you might be able to just have field under one modal (based on just quickly running the application. lmk tho)

Reply:
What do you mean by field?

Reply:
I think I meant everything, not quite sure what i meant there 8 hours ago lol

Reply:
I changed it to just close the modal on submission instead of opening a new one